src/bus/index.spec.js

```diff
@@ -0,0 +1,265 @@
+import { Server, WebSocket } from 'mock-socket';
+import proxyquire from 'proxyquire';
+
+import Channels from './channels';
+import Socket from './socket';
+
+describe('Bus', function() {
+  let baseRequest;
+  let baseSdk;
+  let Bus;
+
+  before(function() {
+    proxyquire.noCallThru();
+    proxyquire.preserveCache();
+
+    Bus = proxyquire('./index', {
+      ws: WebSocket
+    }).default;
+  });
+
+  beforeEach(function() {
+    this.sandbox = sandbox.create();
+
+    baseRequest = {
+      delete: this.sandbox.stub().resolves(),
+      get: this.sandbox.stub().resolves(),
+      post: this.sandbox.stub().resolves(),
+      put: this.sandbox.stub().resolves()
+    };
+
+    baseSdk = {
+      config: {
+        audiences: {
+          bus: fixture.build('audience')
+        }
+      }
+    };
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  describe('constructor', function() {
+    let bus;
+
+    beforeEach(function() {
+      bus = new Bus(baseSdk, baseRequest);
+    });
+
+    it('sets a base url for the class instance', function() {
+      expect(bus._baseUrl).to.equal(`${baseSdk.config.audiences.bus.host}`);
+    });
+
+    it('appends the supplied request module to the class instance', function() {
+      expect(bus._request).to.equal(baseRequest);
+    });
+
+    it('appends the supplied sdk to the class instance', function() {
+      expect(bus._sdk).to.equal(baseSdk);
+    });
+
+    it('appends an instance of Fields to the class instance', function() {
+      expect(bus.channels).to.be.an.instanceof(Channels);
+    });
+
+    it("returns a function called 'connect'", function() {
+      expect(bus.connect).to.be.a('function');
+    });
+  });
+
+  describe('connect', function() {
+    let expectedHost;
+    let expectedOrganization;
+    let server;
+
+    beforeEach(function() {
+      expectedHost = `wss://${faker.internet.domainName()}`;
+      expectedOrganization = fixture.build('organization');
+
+      server = new Server(
+        `${expectedHost}/organizations/${expectedOrganization.id}/stream`
+      );
+    });
+
+    afterEach(function() {
+      server.stop();
+    });
+
+    context('when a websocket already exists for the organization', function() {
+      let bus;
+      let expectedSocket;
+      let promise;
+      let sdk;
+      let socket;
+
+      beforeEach(function() {
+        sdk = {
+          ...baseSdk,
+          auth: {
+            ...baseSdk.auth,
+            getCurrentApiToken: this.sandbox.stub().resolves()
+          }
+        };
+
+        bus = new Bus(sdk, baseRequest);
+        bus._baseSocketUrl = expectedHost;
+
+        socket = new WebSocket(
```

Comment:
Why not just create this in the `new Socket`? You're not using it anywhere else and you are even redeclaring the variable in the .then